api/utils/common.js

```diff
@@ -677,60 +677,145 @@ common.getISOWeeksInYear = function(year) {
 * @param {string} argProperties.has-char - should string property has any latin character in it
 * @param {string} argProperties.has-upchar - should string property has any upper cased latin character in it
 * @param {string} argProperties.has-special - should string property has any none latin character in it
-* @returns {object|false} validated args or false if args do not pass validation
+* @param {boolean} returnErrors - return error details as array or only boolean result
+* @returns {object} validated args in obj property, or false as result property if args do not pass validation and errors array
 */
-common.validateArgs = function(args, argProperties) {
+common.validateArgs = function(args, argProperties, returnErrors) {
 
-    var returnObj = {};
+    if (arguments.length === 2) {
+        returnErrors = false;
+    }
+
+    var returnObj;
+
+    if (returnErrors) {
+        returnObj = {
+            result: true,
+            errors: [],
+            obj: {}
+        };
+    }
+    else {
+        returnObj = {};
+    }
 
     if (!args) {
-        return false;
+        if (returnErrors) {
+            returnObj.result = false;
+            returnObj.errors.push("Missing 'args' parameter.");
+            return returnObj;
+        }
+        else {
+            return false;
```

Comment:
If the validateArgs function returns false (i.e. a boolean value) - 
var validation = common.validateArgs(params.qstring.args, argProps, false); 
//suppose validation is false
if (!(args = validation.obj)) { //returns true
   log.d('Not enough params to create message: %j', params.qstring.args);
   common.returnMessage(params, 400, 'Errors: ' + validation.errors); // output - 'Errors: undefined'
   return;
}

When we are not displaying proper messages to the users, should we show other text. Because, the api response message 'Errors: undefined' is not quite clear.